src/cert_context.rs

```diff
@@ -228,7 +264,48 @@ impl CertContext {
         }
     }
 
-    fn get_encoded_bytes(&self) -> &[u8] {
+    /// Returns the valid uses for this certificate
+    pub fn valid_uses(&self) -> io::Result<ValidUses> {
```

Comment:
It might make sense to return an `Option<Vec<String>>`?

Reply:
I'm not sure that works. There are 4 return types - API failure, No uses valid, All uses valid, and a list of valid uses. I don't want to make `None` mean "API failure or all uses valid." (I found the underlying API very confusing to understand because `.cUsageIdentifier==0` means one of "none" or "all" depending on the value of `GetLastError()`.)

I suppose it could return Result<Option<Vec<String>>> and document that Ok(None) means "all" but I think how it is now is more obvious than that.

How ever you like of course... let me know how you want it.

Reply:
Ok yeah, I guess a custom enum is a bit more clear.